rspirv/sr/constants.rs

```diff
@@ -14,102 +14,21 @@
 
 use crate::spirv;
 
-use super::{Token, Type};
+use super::Token;
 
-/// The class to represent a SPIR-V constant.
+/// Represents a SPIR-V constant.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Constant {
-    pub(in crate::sr) c: ConstantEnum,
-}
-
-#[derive(Clone, Debug, PartialEq)]
-pub(in crate::sr) enum ConstantEnum {
+pub enum Constant {
     Bool(bool),
-    I32(i32),
-    U32(u32),
-    F32(f32),
+    // TODO: Change to arbitrary width datatype
```

Comment:
There aren't any arbitrary sized floating-point libraries on crates as far as I can tell, so we may end up having to roll our own. We could just make the data opaque but people will almost certainly want to be able to modify it.